lib/kvdb/kvdb_rest.c

```diff
@@ -283,14 +284,32 @@ rest_kvdb_params_get(
         free(buf);
     } else {
         char *data;
-        cJSON *root;
+        struct kvdb_cparams cparams;
+        cJSON *merged, *cp_json, *rp_json;
+
+        err = ikvdb_cparams(kvdb, &cparams);
+        assert(err == 0);
 
-        root = kvdb_rparams_to_json(ikvdb_rparams(kvdb));
-        if (ev(!root))
+        cp_json = kvdb_cparams_to_json(&cparams);
+        if (ev(!cp_json))
             return REST_STATUS_INTERNAL_SERVER_ERROR;
 
-        data = (pretty ? cJSON_Print : cJSON_PrintUnformatted)(root);
-        cJSON_Delete(root);
+        rp_json = kvdb_rparams_to_json(ikvdb_rparams(kvdb));
+        if (ev(!rp_json)) {
+            cJSON_Delete(cp_json);
+            return REST_STATUS_INTERNAL_SERVER_ERROR;
+        }
+
+        merged = cJSONUtils_MergePatchCaseSensitive(cp_json, rp_json);
+        if (ev(!merged)) {
+            cJSON_Delete(cp_json);
+            cJSON_Delete(rp_json);
+            return REST_STATUS_INTERNAL_SERVER_ERROR;
+        }
+
+        data = (pretty ? cJSON_Print : cJSON_PrintUnformatted)(merged);
+        cJSON_Delete(merged);
+        cJSON_Delete(rp_json);
```

Comment:
Are you missing cJSON_Delete(cp_json) here?

Reply:
No, cJSON is not a great API. `merged` is `cp_json` with all the children of `rp_json` duplicated into it.